Pathogens/Utility/location_utils/location_functions.py

```diff
@@ -178,21 +178,51 @@ def compute_pin_info(df: pd.DataFrame, geo_dfs: dict) -> pd.DataFrame:
 
     return df
 
-def get_lng_lat(geocoder_search_text, geocoder_data_type, country_code=None):
-    # If a city doesn't have a state associated with it,
-    # we cannot accurately find its location
-    if (not geocoder_search_text) or (geocoder_data_type == 'place' and "," not in geocoder_search_text):
+def get_mapbox_api_query_url(geocoder_search_text, geocoder_data_type, country_code):
+    return f"https://api.mapbox.com/geocoding/v5/mapbox.places/{geocoder_search_text}.json?" \
+          f"access_token={os.getenv('MAPBOX_API_KEY')}&types={geocoder_data_type}&country={country_code}"
+
+def parse_mapbox_response(response):
+    data = response.json()
+    if data and "features" in data and len(data['features']) > 0:
+        return data['features'][0]['center']
+    else:
         return None
 
-    url = f"https://api.mapbox.com/geocoding/v5/mapbox.places/{geocoder_search_text}.json?" \
-          f"access_token={os.getenv('MAPBOX_API_KEY')}&types={geocoder_data_type}"
+def get_city_lat_lng(city_name, state_name, country_name):
```

Comment:
Kind of confusing but the story of this file is basically that I set up like a whole bunch of fallbacks because I kept finding these edge cases in our data where the city name would be the name of the state or the city name would be there and the state name wouldn't be there, etc.

With all of these fallbacks literally every pin has a lat and lng and I did a really good search through the pins to see if I could find any in the wrong country and I only found one which was on the wrong island in Indonesia but that's literally it so I'm pretty confident this is good. My only thought for a future enhancement here would maybe be an override json object where we can put specific locations where we know mapbox has bad data for (so far just literally this one spot in Indonesia) and then correct the latlngs ourselves but I can take that on later, let's just fix 99.9% of these first :)

Reply:
Ok sounds good! And also just a note on being in the wrong island, that is fine we only really care about country level correctness AFAIK and will probably have to introduce some noise on purpose to ensure all pins are visible (multiple pins in the same city would overlap otherwise)